metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -519,6 +524,30 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
             self.report.set_project_state(project_id.id, "Metadata Extraction")
             yield from self._process_project(conn, project_id)
 
+        if self.config.include_table_lineage:
+            if (
+                self.config.store_last_lineage_extraction_timestamp
+                and self.redundant_run_skip_handler.should_skip_this_run(
+                    cur_start_time_millis=datetime_to_ts_millis(self.config.start_time)
+                )
+            ):
+                # Skip this run
+                self.report.report_warning(
+                    "lineage-extraction",
+                    f"Skip this run as there was a run later than the current start time: {self.config.start_time}",
+                )
+                return
```

Comment:
The state handlers /really/ confuse me (what are these timestamps it's keeping track of? Why are we storing one per project per lineage / usage lookup??), so I just left this in here. It feels like we should only have to calculate this once for all projects, for lineage, and ideally, for lineage + usage.

Reply:
Right, lineage state needs to be updated for entire ingestion run only once and not per project. I've fixed this for you in lineage.

Reply:
This is correct, you are right.